swap.go

```diff
@@ -39,19 +42,40 @@ func newSwapKit(hash lntypes.Hash, swapType swap.Type, cfg *swapConfig,
 	}
 
 	return &swapKit{
-		swapConfig: *cfg,
-		hash:       hash,
-		log:        log,
-		state:      loopdb.StateInitiated,
-		contract:   contract,
-		swapType:   swapType,
+		swapConfig:      *cfg,
+		hash:            hash,
+		log:             log,
+		state:           loopdb.StateInitiated,
+		contract:        contract,
+		swapType:        swapType,
+		protocolVersion: protocolVersion,
 	}
 }
 
+// GetHtlcScriptVersion returns the correct HTLC script version for the passed
+// protocol version.
+func GetHtlcScriptVersion(
+	protocolVersion loopdb.ProtocolVersion) swap.ScriptVersion {
+
+	if protocolVersion != loopdb.ProtocolVersionUnrecorded &&
+		protocolVersion >= loopdb.ProtocolVersionHtlcV2 {
+
+		// Use HTLC v2 script only if we know the swap was initiated
+		// with a client that supports HTLC v2. Unrecorded protocol
+		// version implies that there was no protocol version stored
+		// along side a serialized swap that we're resuming in which
+		// case the swap was initiated with HTLC v1 script.
+		return swap.HtlcV2
+	}
+
+	return swap.HtlcV1
+}
+
 // getHtlc composes and returns the on-chain swap script.
 func (s *swapKit) getHtlc(outputType swap.HtlcOutputType) (*swap.Htlc, error) {
 	return swap.NewHtlc(
-		swap.HtlcV1, s.contract.CltvExpiry, s.contract.SenderKey,
+		GetHtlcScriptVersion(s.contract.ProtocolVersion),
```

Comment:
It looks like the quote call still always uses htlc v1. I don't think it matters much, but then we need to change it when we eventually phase out v1.

Reply:
Indeed, fixed.